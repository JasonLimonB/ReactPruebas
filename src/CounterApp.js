import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ( {value} ) => {
    
    
    const [ counter, setCounter ] = useState( value );
 
    const sumar = ()=>{
        setCounter( counter + 1 );
    }
    const restar = () =>{
        setCounter( counter - 1 );
    }
    const reset = () =>{
        setCounter( value );
    }

    return (
        <>
            <h1>Counter app</h1>
            <p id="contador">{ counter }</p>
            <button onClick={sumar} className="btn"> Sumar </button>
            <button onClick={restar} className="btn"> Restar </button>
            <button onClick={reset} className="btn"> Reset </button>
        </>
    );
}

CounterApp.protoTypes = {
    value: PropTypes.number
}

export default CounterApp;

