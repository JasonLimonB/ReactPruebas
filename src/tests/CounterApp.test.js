import React from 'react';
import '@testing-library/dom';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

describe('Test de componetn <CounterApp />', () => {
   
    

    test('realizando prueba de props ', () => {

        const wrapper = shallow( <CounterApp /> );

        expect( wrapper ).toMatchSnapshot();

    });

    test('mandando el valor a la prop ', () => {
        
        const wrapper = shallow( <CounterApp value={10} /> );
        const couneter = wrapper.find('#contador').text().trim();
        
        expect( couneter ).toBe('10');

    });

    // Ahora en adelante las pruebas van a ser para simular click en los botones
    // se tomara por la posicion de los botones ya que no tienen un id para identificarlos

    test('simulando click +1', () => {
        const wrapper = shallow(<CounterApp value={10}/>);
        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('#contador').text().trim();

        expect( counterText ).toBe('11');

    });

    // Realizar la prueba del boton de restar
    test('simalacion de click -1', () => {
        const wrapper = shallow(<CounterApp value={10}/>);
        wrapper.find('button').at(1).simulate('click');

        const counterText = wrapper.find('#contador').text().trim();

        expect( counterText ).toBe('9');

    });


    // Ahora la prueba serÃ¡ para resetear al valor indicado y haga su funcion
    test('debe resetear el valor', () => {
        const wrapper = shallow(<CounterApp value={10}/>);
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(2).simulate('click');

        const counterText = wrapper.find('#contador').text().trim();

        expect( counterText ).toBe( '10' );

    });

});
