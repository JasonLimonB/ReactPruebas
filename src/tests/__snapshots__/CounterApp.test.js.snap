// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de componetn <CounterApp /> realizando prueba de props  1`] = `
<Fragment>
  <h1>
    Counter app
  </h1>
  <p
    id="contador"
  />
  <button
    className="btn"
    onClick={[Function]}
  >
     Sumar 
  </button>
  <button
    className="btn"
    onClick={[Function]}
  >
     Restar 
  </button>
  <button
    className="btn"
    onClick={[Function]}
  >
     Reset 
  </button>
</Fragment>
`;
